client/web/src/repo/RepoRevisionSidebarCommits.tsx

```diff
@@ -67,83 +67,94 @@ interface Props extends Partial<RevisionSpec>, FileSpec {
-    const queryCommits = useCallback(
-        (args: { query?: string }): Observable<CommitAncestorsConnectionFields> =>
-            fetchCommits(props.repoID, props.revision || '', { ...args, currentPath: props.filePath || '' }),
-        [props.repoID, props.revision, props.filePath]
-    )
+    const { connection, error, loading, hasNextPage, fetchMore } = useShowMorePagination<
+        FetchCommitsResult,
+        FetchCommitsVariables,
+        GitCommitFields
+    >({
+        query: FETCH_COMMITS,
+        variables: {
+            afterCursor: null,
+            first: props.defaultPageSize || 100,
+            query: '',
+            repo: props.repoID,
+            revision: props.revision || '',
+            currentPath: props.filePath || '',
+        },
+        getConnection: result => {
+            const { node } = dataOrThrowErrors(result)
+
+            if (!node) {
+                return { nodes: [] }
+            }
+            if (node.__typename !== 'Repository') {
+                return { nodes: [] }
+            }
+            if (!node.commit?.ancestors?.nodes) {
+                return { nodes: [] }
+            }
+
+            return node.commit.ancestors
+        },
+        options: {
+            // Currently "after" is used as a commit filtering option to return
+            // commits after a specific date. Currently the pagination is
+            // implemented by using afterCursor instead and setting this boolean
+            // will ensure that the pagination works correctly.
+            customAfterCursor: true,
+            fetchPolicy: 'cache-first',
+        },
+    })
 
     return (
-        <FilteredConnection<
-            GitCommitFields,
-            Pick<CommitNodeProps, 'location' | 'preferAbsoluteTimestamps'>,
-            CommitAncestorsConnectionFields
-        >
-            className="list-group list-group-flush"
-            ariaLive="polite"
-            listClassName={styles.list}
-            summaryClassName={styles.summary}
-            loaderClassName={styles.loader}
-            compact={true}
-            noun="commit"
-            pluralNoun="commits"
-            queryConnection={queryCommits}
-            nodeComponent={CommitNode}
-            nodeComponentProps={{ location: props.location, preferAbsoluteTimestamps: props.preferAbsoluteTimestamps }}
-            defaultFirst={props.defaultPageSize || 100}
-            hideSearch={true}
-            useURLQuery={false}
-            history={props.history}
-            location={props.location}
-        />
+        <ConnectionContainer>
+            {error && <ErrorAlert error={error} />}
+            {connection &&
+                connection.nodes.map(node => (
+                    <CommitNode
+                        node={node}
+                        location={props.location}
+                        preferAbsoluteTimestamps={props.preferAbsoluteTimestamps}
+                    />
```

Comment:
The CommitNode must have `key` so React knows how to diff this properly. Not sure if this field already has an ID but if not use _anything_ that is unique within this list. It does not need to be globally unique. 

```suggestion
                connection.nodes.map(node => (
                    <CommitNode
                        key={node.id}
                        node={node}
                        location={props.location}
                        preferAbsoluteTimestamps={props.preferAbsoluteTimestamps}
                    />
```